toolchain/parse/handle_binding_pattern.cpp

```diff
@@ -77,6 +73,14 @@ auto HandleBindingPattern(Context& context) -> void {
                      : StateKind::BindingPatternFinishAsGeneric;
     // Use the `:` or `:!` for the root node.
     state.token = context.Consume();
+
+    if (token_kind == Lex::TokenKind::ColonExclaim) {
+      // A virtual node to come before the compile time binding's type
```

Comment:
Grammar feels a little odd here, WDYT of:

```suggestion
      // Add a virtual node before the compile time binding's type
```